cylc/flow/scripts/validate_reinstall.py

```diff
@@ -135,12 +137,18 @@ def vro_cli(parser: COP, options: 'Values', workflow_id: str):
     try:
         detect_old_contact_file(workflow_id)
     except ServiceFileError:
-        # Workflow is definately running:
+        # Workflow is definitely running:
         workflow_running = True
     else:
-        # Workflow is definately stopped:
+        # Workflow is definitely stopped:
         workflow_running = False
 
+    # We won't be able to restart stopped 7.x workflows. Check for this NOW:
+    if not workflow_running:
```

Comment:
From experimentation we don't get this far for Cylc 7 workflows because `detect_old_contact_file` fails.

Methodology to test:

```console
$ # Write a Cylc 7 compatible workflow in `~/cylc-src`.

$ # install it with Cylc 8
$ cylc install .

$ # make it look like it *wasn't* installed with Cylc 8
$ (cd ~/cylc-run/<id>; mv _cylc-install _cylc-install-2)

$ # run it with Cylc 7
$ CYLC_VERSION=7 cylc run <id>` --hold

$ # make it look like it *was* installed with Cylc 8
$ (cd ~/cylc-run/<id>; mv _cylc-install-2 _cylc-install)

$ # make it look like there are changes to reload & vr'em
$ echo -e '\n' >>> suite.rc
$ cylc vr <id>
```

Reply:
Suggestion:

```python
try:
    detect_old_contact_file(workflow_id)
except ServiceFileError:
    ...
except Exception:
    if looks_cylc_7ey():
        raise CylcError('Cylc 8 cannot restart or reload Cylc 7 workflows')
```

Where:

```
def looks_cylc_7ey():
    with open(path_to_contact_file, 'r') as f:
        for line in f:
            if 'CYLC_VERSION=7' in line:
                return True
    return False
```

---

Or simply:


```python
try:
    detect_old_contact_file(workflow_id)
except ServiceFileError:
    ...
except Exception:
    LOG.critical(
        'Cannot tell if the workflow is running'
        '\nNote, Cylc 8 cannot restart Cylc 7 workflows.'
    )
    raise
```